### ISSUE TITLE
Forward with Template doesn't send off mail automatically any more

### ISSUE THREAD

**Issue by RealRaven2000** [RealRaven2000/SmartTemplates#231]
Presumably since version 102.9, the FiltaQuilla action "forward with smart template" doesn't  automatically send off the generated email  anymore. The composer window is opened correctly and all fields are processed, but the user has to send off the email manually.

**Comment by RealRaven2000** [comment]
Just debugging this at the moment, I see that the global variable `SmartTemplate4.PreprocessingFlags` gets messed up in the function `insertTemplate()` when this is called:

```js
await SmartTemplate4.Util.resolveDeferredBatch(gMsgCompose.editor);
```

`PreprocessingFlags` is an object that contains the flag for automatic sending. somehow it gets converted to an empty array at this stage...

**Comment by RealRaven2000** [comment]
Found the cause of problem at the end of `SmartTemplate4.Util.resolveDeferredBatch()` :

https://github.com/RealRaven2000/SmartTemplates/blob/9ffa468c9d6cd965371513c295f7ea2ef9ca4d04/chrome/content/smartTemplate-util.js#L1652

this is meant to reset the array of modified headers used, instead it overwrites `PreprocessingFlags` ! Patched version will be released shortly!



```js
// old code:
SmartTemplate4.PreprocessingFlags= []; 
// new code:
SmartTemplate4.PreprocessingFlags.modifiedHeaders = []; 
```

**Comment by RealRaven2000** [comment]
Fixed version below. (To do: update version notes on splash screen)

[smartTemplate-fx-3.16.1pre7.zip](https://github.com/RealRaven2000/SmartTemplates/files/11017866/smartTemplate-fx-3.16.1pre7.zip)

To test this version, download the zip file and then drag it into Thunderbird Add-ons Manager. Do **not extract contents** to install.

**Comment by RealRaven2000** [comment]
I also noticed that SmartTemplates does not tidy up the formatted fields before sending. This is caused by the function `SmartTemplate4.Util.cleanupDeferredFields()` now being asynchronous. The used `compose-send-message` event  cannot await an async function when it is being dispatched. I will therefore add code that resolves all variables before SmartTemplates arms the timer to send off the email.

See fixed version below:

[smartTemplate-fx-3.16.1pre9.zip](https://github.com/RealRaven2000/SmartTemplates/files/11028427/smartTemplate-fx-3.16.1pre9.zip)

To test this version, download the zip file and then drag it into Thunderbird Add-ons Manager. Do **not extract contents** to install.

**Comment by RealRaven2000** [comment]
Fixed in release 3.16.1 - published 21/03/2023
